lib/react_client.dart

```diff
@@ -614,11 +614,19 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
       });
 
   void handleComponentDidCatch(Component2 component, dynamic error, ReactErrorInfo info) => zone.run(() {
-        component.componentDidCatch(error, info);
+        try {
+          rethrowError(error);
```

Comment:
Should we add any comments to explain why we're doing this? Some may be able to reason about it, but it seems a little unintuitive.